server/container/guice/cassandra-rabbitmq-ldap-guice/src/test/java/org/apache/james/CassandraRabbitMQLdapJamesServerTest.java

```diff
@@ -0,0 +1,121 @@
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Duration.ONE_HUNDRED_MILLISECONDS;
+
+import java.io.IOException;
+
+import org.apache.commons.net.imap.IMAPClient;
+import org.apache.james.core.Domain;
+import org.apache.james.modules.RabbitMQExtension;
+import org.apache.james.modules.SwiftBlobStoreExtension;
+import org.apache.james.modules.TestJMAPServerModule;
+import org.apache.james.modules.protocols.ImapGuiceProbe;
+import org.apache.james.modules.protocols.SmtpGuiceProbe;
+import org.apache.james.utils.IMAPMessageReader;
+import org.apache.james.utils.SMTPMessageSender;
+import org.apache.james.utils.SpoolerProbe;
+import org.awaitility.Awaitility;
+import org.awaitility.Duration;
+import org.awaitility.core.ConditionFactory;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+class CassandraRabbitMQLdapJamesServerTest {
+    private static final int LIMIT_TO_10_MESSAGES = 10;
+    private static final String JAMES_USER = "james-user";
+    private static final String PASSWORD = "secret";
+    private static Duration slowPacedPollInterval = ONE_HUNDRED_MILLISECONDS;
+
+    private static ConditionFactory calmlyAwait = Awaitility
+        .with().pollInterval(slowPacedPollInterval)
+        .and().with().pollDelay(slowPacedPollInterval)
+        .await();
+
+    interface MailsShouldBeWellReceived {
+        @RegisterExtension
+        IMAPMessageReader imapMessageReader = new IMAPMessageReader();
+        SMTPMessageSender messageSender = new SMTPMessageSender(Domain.LOCALHOST.asString());
+
+        @Test
+        default void mailsShouldBeWellReceived(GuiceJamesServer server) throws Exception {
+            messageSender.connect(JAMES_SERVER_HOST, server.getProbe(SmtpGuiceProbe.class).getSmtpPort())
+                .sendMessage("bob@any.com", JAMES_USER + "@localhost");
+
+            calmlyAwait.until(() -> server.getProbe(SpoolerProbe.class).processingFinished());
+
+            imapMessageReader.connect(JAMES_SERVER_HOST, server.getProbe(ImapGuiceProbe.class).getImapPort())
+                .login(JAMES_USER, PASSWORD)
+                .select("INBOX")
+                .awaitMessage(calmlyAwait);
+        }
+    }
+
+    interface UserFromLdapShouldLogin {
+
+        @Test
+        default void userFromLdapShouldLoginViaImapProtocol(GuiceJamesServer server) throws IOException {
+            IMAPClient imapClient = new IMAPClient();
+            imapClient.connect(JAMES_SERVER_HOST, server.getProbe(ImapGuiceProbe.class).getImapPort());
+
+            assertThat(imapClient.login(JAMES_USER, PASSWORD)).isTrue();
+        }
+    }
+
+    interface ContratSuite extends JmapJamesServerContract, MailsShouldBeWellReceived, UserFromLdapShouldLogin {}
+
+    @Nested
+    @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+    class WithSwift implements ContratSuite {
```

Comment:
Probably you are running `WithSwift` and `WithoutSwift` with two different docker bootstraps(cassandra+rabbit+swift+es up and down for executing `WithSwift` and again for `WithoutSwift`). I guess we could use once to save execution time of this test class.

Reply:
We're now having `SwiftDockerSingleton` and `CassandraDockerSingleton`, `RabbitMQDockerSingleton` is ahead in the Jmap tests PR, you can wait till it is merged, then apply rabbit singleton to `RabbitMQExtension`. Now you are free to make `LdapDockerSingleton`

Reply:
I implemented